libs/perun/services/src/lib/store.service.ts

```diff
@@ -62,73 +62,41 @@ export class StoreService {
   }
 
   getProperty<T extends keyof PerunConfig>(key: T): PerunConfig[T] {
+    if (!this.instanceConfig || !this.defaultConfig) {
+      return null;
+    }
+
     const configs: PerunConfig[] = [
       this.instanceConfig?.brandings?.[this.branding],
       this.instanceConfig,
-      this.defaultConfig,
     ];
 
+    const defaultValue: PerunConfig[T] = this.defaultConfig[key];
     let currentValue: PerunConfig[T] = null;
     for (const config of configs) {
       if (config && (currentValue === null || currentValue === undefined)) {
         currentValue = config[key];
       }
     }
 
-    return currentValue;
-  }
-
-  /**
-   * @deprecated - Use method `getProperty` instead.
-   */
-  /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment */
-  get(...keys: string[]): any {
-    let currentValue: string;
-
-    if (this.branding !== '') {
-      const brandingConfig: PerunConfig = this.instanceConfig.brandings[this.branding];
-      for (let i = 0; i < keys.length; ++i) {
-        if (i === 0) {
-          currentValue = brandingConfig[keys[i]];
-        } else {
-          if (currentValue === undefined) {
-            break;
-          }
-          currentValue = currentValue[keys[i]];
-        }
-      }
+    if (!currentValue) {
+      return defaultValue;
     }
 
-    if (this.instanceConfig !== undefined && currentValue === undefined) {
-      for (let i = 0; i < keys.length; ++i) {
-        if (i === 0) {
-          currentValue = this.instanceConfig[keys[i]];
-        } else {
-          if (currentValue === undefined) {
-            break;
-          }
-          currentValue = currentValue[keys[i]];
-        }
-      }
-    }
+    return this.addMissingValuesToProperty(currentValue, defaultValue);
+  }
 
-    if (this.defaultConfig === undefined) {
-      return undefined;
-    }
-    if (currentValue === undefined) {
-      for (let i = 0; i < keys.length; ++i) {
-        if (i === 0) {
-          currentValue = this.defaultConfig[keys[i]];
-        } else {
-          if (currentValue === undefined) {
-            // console.error('Missing value in default config: ' + keys);
-            break;
-          }
-          currentValue = currentValue[keys[i]];
-        }
+  addMissingValuesToProperty<T extends keyof PerunConfig>(
+    value: PerunConfig[T],
+    defaultValue: PerunConfig[T]
+  ): PerunConfig[T] {
+    if (typeof value === 'object' && !Array.isArray(value) && value) {
```

Comment:
There is a little "problem" with an array of objects properties like the `custom_labels` property. On the other hand properties of this type are just about labels (`ProfileAttribute[]`, `ProfileExtService[]` and `ProfileCustomLabel[]`) so I don't think that it is a big problem at all. IMHO it's more about a bad configuration, if someone put just a part of this config type to the `instanceConfig`, what do you think (@xkostka2)?

Reply:
I agree, default values in array-of-objects properties are usually just examples of how to fill those properties.